src/utils.ts

```diff
@@ -131,3 +132,26 @@ export function unrefComponent(components: any, ref: string): any {
   const [type, object] = ref.match(/(?<=\[")([^"]+)/g) as string[];
   return components[type][object];
 }
+
+function resolveDocumentReference<T>(document: any, reference: string): T | undefined {
+  if (reference[0] === "#") {
+    const parts = reference.replace(/^#\//, "").split("/");
+    const result = _.get(document, parts.join("."));
```

Comment:
I’d like to not add `lodash` to this library all for one function. Why not use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) instead?

Reply:
I've change to the single `get` function export of lodash.  

I'd rather not write some code to dynamically walk a JS object when lodash already has a function that already has it and is well tested.

Reply:
I know lodash isn’t gigantic, but aren’t you just using optional chaining here? It’s more about adding weight for something built into the language now.

Reply:
Considering the paths are dynamic based on the contents of the ref, could you show me an example, I don't think optional chaining does what I think lodash's get method does.

It seems to me that together we are going going to iterate the parts of the path, with a current value then traverse down using each property name.  Optional chaining does not do that unless you write the property path explicitly in the code which we can't do because references are dynamic in both length and content.

Reply:
haven't looked at the details of the above, but you can use expressions with optional chaining:

```ts
const foo = 'x-foo'

obj?.[foo]?.['bar'].x
```

not sure how I'd approach the length, possibly with a loop ...